pkg/webhooks/validation/server/server_test.go

```diff
@@ -223,6 +251,52 @@ func TestAdmitPilot(t *testing.T) {
 			}
 		})
 	}
+
+	smeV1 := makeServiceMeshExtensionConfig(t, "v1")
+	smeV1alpha1 := makeServiceMeshExtensionConfig(t, "v1alpha1")
+	deprecatedWarning := "ServiceMeshExtension is deprecated. Please use WasmPlugin instead."
+
+	maistraCases := []struct {
+		name    string
+		admit   admitFunc
+		in      *kube.AdmissionRequest
+		allowed bool
+		warning string
+	}{
+		{
+			name:  "v1 ServiceMeshExtension deprecation warning",
+			admit: wh.validate,
+			in: &kube.AdmissionRequest{
+				Kind:      kubeApisMeta.GroupVersionKind{Kind: "ServiceMeshExtension"},
+				Object:    runtime.RawExtension{Raw: smeV1},
+				Operation: kube.Create,
+			},
+			allowed: true,
+			warning: deprecatedWarning,
+		},
+		{
+			name:  "v1 ServiceMeshExtension deprecation warning",
+			admit: wh.validate,
+			in: &kube.AdmissionRequest{
+				Kind:      kubeApisMeta.GroupVersionKind{Kind: "ServiceMeshExtension"},
+				Object:    runtime.RawExtension{Raw: smeV1alpha1},
+				Operation: kube.Create,
+			},
+			allowed: true,
+			warning: deprecatedWarning,
+		},
+	}
```

Comment:
1. Operation `kube.Update` is not covered.
2. Both test cases have the same `admit`, `allowed` and `warning` fields, so the test case struct might be simplified to:
```
name       string
smeVersion []byte
operation  string
```
Then you would call in the test:
```
got := wh.validate(&kube.AdmissionRequest{
  Kind:      kubeApisMeta.GroupVersionKind{Kind: "ServiceMeshExtension"},
  Object:    runtime.RawExtension{Raw: c.smeVersion},
  Operation: operation,
})
```
It's not very important, but I just don't see a need to parameterize constant values.

Reply:
Done